jsonnet/thanos-ruler.jsonnet

```diff
@@ -114,6 +129,52 @@ local authorizationRole =
         secret.mixin.metadata.withNamespace(thanosRulerConfig.namespace) +
         secret.mixin.metadata.withLabels({ 'k8s-app': 'thanos-ruler' }),
 
+      // query config which holds http configuration
+      // for communication between thanos ruler and thanos qurier.
+      alertmanagersConfigSecret:
+        local alertmanagerConfig = {
+          'alertmanagers.yaml': std.manifestYamlDoc({
+            alertmanagers: [{
+              http_config: {
+                bearer_token_file: '/var/run/secrets/kubernetes.io/serviceaccount/token',
+                tls_config: {
+                  ca_file: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
+                  server_name: 'alertmanager-main.openshift-monitoring.svc',
```

Comment:
What happens if the service CA bundle rotates, aka the alertmanager TLS endpoint gets a new server certificate? Does Thanos Ruler happen to handle this rotation in-process or do we have to rotate it manually in cluster-monitoring-operator? @lilic cc @bwplotka

Reply:
Thanos components use https://godoc.org/github.com/prometheus/common/config#NewClientFromConfig under the hood and the returned `http.Client` detects whenever the certificates are rotated.

Reply:
suggestion: let's add a comment in `manifests.go` that this is handled in-process.

Reply:
Where in manifest.go?